### PR TITLE
Fix RBAC migration SQLServer cascade problem

### PR DESCRIPTION
**Pull request by fourteenmeister**
Need testing.
I tested on SQL Server 2008.

### CODE REVIEW

**Comment by samdark** [comment]
Which problem it fixes?

**Comment by samdark** [comment]
https://github.com/yiisoft/yii2/issues/6876

**Comment by samdark** [comment]
@thejahweh, @fourteenmeister could you update it so it could be applied to latest master code?

File: framework/rbac/migrations/m140506_102106_rbac_init.php [hunk 1]
```diff
@@ -17,6 +17,15 @@
  */
 class m140506_102106_rbac_init extends \yii\db\Migration
 {
+    
+    protected $isMsSQL;
+
+    public function init()
+    {
+        parent::init(); // TODO: Change the autogenerated stub
+        $this->isMsSQL = $this->db->driverName == 'mssql' || $this->db->driverName == 'sqlsrv' || $this->db->driverName == 'dblib';
```
Comment (samdark) [inline framework/rbac/migrations/m140506_102106_rbac_init.php]: Better to be done via function like

```php
private function isMSSQL()
    {
        return $this->db->driverName === 'mssql' || $this->db->driverName === 'sqlsrv' || $this->db->driverName === 'dblib';
    }
```

File: framework/rbac/migrations/m140506_102106_rbac_init.php [hunk 2]
```diff
@@ -77,6 +91,43 @@ public function up()
...
+                ELSE
+                    BEGIN
+                        DELETE FROM dbo.{$authManager->itemChildTable} WHERE parent IN (SELECT name FROM deleted) OR child IN (SELECT name FROM deleted);
+                        DELETE FROM dbo.{$authManager->itemTable} WHERE name IN (SELECT name FROM deleted);
+                    END
+            END;");
+        }
     }
 
     public function down()
```
Comment (samdark) [inline framework/rbac/migrations/m140506_102106_rbac_init.php]: Shouldn't we drop trigger here?

**Comment by thejahweh** [comment]
sorry @samdark but I just don't know how to update a pull-request, there is no branch.

**Comment by samdark** [comment]
Weird. Creating new one is a good option as well.

**Pull request closed without merging**